code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+		for(var/i in rocket_stop)
+			if(istype(j, i))
+				skip_obj = TRUE
+				break
+		if(skip_obj)
+			continue
+		if(istype(j, /obj/machinery/constructable_frame))
+			qdel(j)
+		else if(istype(j, /obj/structure/grille/broken))
+			continue
+		else if(istype(j, /obj/structure/table_frame))
+			continue
+		else if(istype(j, /obj/structure))
+			if(j.density)
+				qdel(j)
+
+	return TRUE
+	
+/obj/item/projectile/a66mm/on_hit(turf/target, blocked = 0)
+	if(locate(/obj/mecha) in target.contents)
+		for(var/obj/mecha/M in target.contents)
+			if(!is_fellow_antag(M) && M.density)
+				M.take_damage(damage, BRUTE)
+				break
+	else if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+	else if(istype(target, /turf/simulated/wall/r_wall))
+		var/turf/simulated/wall/r_wall/W = target
+		W.add_dent(WALL_DENT_HIT)
+		target = get_turf(src)
+	else if(istype(target, /turf/simulated/wall))
+		var/turf/simulated/wall/W = target
+		W.break_wall()
+		target = get_turf(src)
+	else if(locate(/obj/structure/girder) in target.contents)
+		var/obj/structure/girder/G = locate(/obj/structure/girder) in target.contents
+		G.deconstruct(FALSE)
+		target = get_turf(src)
+	else
+		target = get_turf(src)
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	explosion(target, -1, -1, ex_range+1, cause = "[type] fired by [key_name(firer)]")
+	var/datum/effect_system/explosion/smoke/E = new/datum/effect_system/explosion/smoke()
+	E.set_up(target)
+	E.start()
+	return TRUE
+
+/obj/item/projectile/a66mm/on_range()
+	on_hit(get_turf(src))
+	..()
+
+/obj/item/projectile/a66mm/rod
+	name ="66mm Rod rocket"
+	desc = "A rocket designed to punch through almost anything."
+	icon_state = "SLRW-rocket"
+	alwayslog = FALSE
+	rocket_stop = null
+	ex_range = 1
+
+/obj/item/projectile/a66mm/rod/Initialize()
+	..()
+	rocket_stop -= /turf/simulated/wall
+	rocket_stop -= /obj/structure/girder
+	rocket_stop -= /obj/machinery/door/airlock/highsecurity
+	rocket_stop += /turf/simulated/wall/r_wall
+
+/obj/item/projectile/a66mm/rod/prehit(turf/target as turf)
+	if(!..())
+		return FALSE
+	if(istype(target, /turf/simulated/wall))
+		sfx_hit(60, 40)
+		target.ChangeTurf(/turf/simulated/floor/plating)
+		ffx_hit(target)
+	return TRUE
+
+/obj/item/projectile/a66mm/rod/on_hit(turf/target, blocked = 0)
+	if(locate(/obj/mecha) in target.contents)
+		for(var/obj/mecha/M in target.contents)
+			if(!is_fellow_antag(M) && M.density)
+				M.take_damage(damage, BRUTE)
+				break
+	else if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	for(var/turf/T in range(ex_range, target))
+		ffx_hit(T)
```

Comment:
Shouldn't the explosion handle this on it's own?

Reply:
At ex_act() severity 3, floor tiles have a 50% chance to break and show a visual explosion effect. This makes sure that all tiles show an explosion effect.